Packs/MongoDBAtlas/Integrations/MongoDBAtlas/MongoDBAtlas.py

```diff
@@ -0,0 +1,569 @@
+    new_min_time = last_fetched_event.get('created') if last_fetched_event else None
+    return results, new_min_time
+
+
+def fetch_event_type(client: Client, fetch_limit: int, last_run: dict) -> tuple[list, dict]:
+    """
+    Fetches events until fetch_limit is reached, or no more events are available.
+
+    Args:
+        client (Client): MongoDBAtlas client.
+        fetch_limit (int): The maximum number of events to fetch.
+        last_run (dict): Dictionary containing data from the previous run.
+
+    Returns:
+        - The list containing all fetched events.
+        - The object to save for the next run.
+    """
+    min_date = last_run.get('min_time')
+    events_with_created_min_time = last_run.get('events_with_created_min_time') or []
+
+    if not min_date:  # first time fetching events
+        output, new_min_time = first_time_fetching_events(client, fetch_limit)
+        new_last_run_obj = create_last_run_dict_for_events(output, new_min_time)
+        return output, new_last_run_obj
+
+    demisto.debug(f'Start to fetch events with {min_date}')
+
+    results: dict = client.get_events_request(min_date=min_date)
+    response = get_last_page_of_events(client, results)
+    links = response.get('links', [])
+    events = response.get('results', [])
+
+    current_fetched_events_amount = 0
+    output = []
+    new_min_date = min_date
+
+    while True:
+        for event in reversed(events):
+            event_id = event.get('id')
+            if event_id in events_with_created_min_time:
+                continue
+
+            enrich_event(event, 'events')
+            output.append(event)
+            current_fetched_events_amount += 1
+            new_min_date = get_latest_date(new_min_date, event.get('created'))
+
+            if current_fetched_events_amount >= fetch_limit:
+                demisto.debug(f'Fetch limit reached. Total events fetched: {len(output)}')
+                new_last_run_obj = create_last_run_dict_for_events(output, new_min_date)
+                return output, new_last_run_obj
+
+        previous_page = get_page_url(links, page_type='previous')
+        if previous_page:
+            response = client.get_response_from_page_link(previous_page)
+            events = response.get('results', [])
+            links = response.get('links', [])
+        else:
+            break
+
+    demisto.debug(f'No more events left to fetch. Total events fetched: {len(output)}')
+    new_last_run_obj = create_last_run_dict_for_events(output, new_min_date)
+    return output, new_last_run_obj
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: Client) -> str:
+    """
+    Returning 'ok' indicates that the integration works like it suppose to. Connection to the service is successful.
+
+    Args:
+        client: MongoDB Atlas client
+
+    Returns:
+        'ok' if test passed, anything else will fail the test
+    """
+    client.get_alerts_request(page_num=1, items_per_page=1)
```

Comment:
For the test module, it is better to check the `fetch_events` command (with a limit of 1). The whole purpose of this collector is the fetch, so we want to test it to make sure it works